cli/src/command/generate_config.rs

```diff
@@ -0,0 +1,214 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Standard Exonum CLI command used to generate public and secret config files
+//! of the node using provided common configuration file.
+
+use exonum::{
+    blockchain::ValidatorKeys,
+    crypto::{generate_keys_file, PublicKey},
+};
+use failure::{bail, Error};
+use serde::{Deserialize, Serialize};
+use structopt::StructOpt;
+
+use std::{
+    fs,
+    net::{IpAddr, SocketAddr},
+    path::{Path, PathBuf},
+};
+
+use crate::{
+    command::{ExonumCommand, StandardResult},
+    config::{CommonConfigTemplate, NodePrivateConfig, NodePublicConfig, SharedConfig},
+    io::{load_config_file, save_config_file},
+    password::{PassInputMethod, Passphrase, PassphraseUsage, SecretKeyType},
+};
+
+/// Name for a file containing consensus secret key.
+pub const CONSENSUS_SECRET_KEY_NAME: &str = "consensus.key.toml";
+/// Name for a file containing service secret key.
+pub const SERVICE_SECRET_KEY_NAME: &str = "service.key.toml";
+/// Name for a file containing the public part of the node configuration.
+pub const PUB_CONFIG_FILE_NAME: &str = "pub.toml";
+/// Name for a file containing the secret part of the node configuration.
+pub const SEC_CONFIG_FILE_NAME: &str = "sec.toml";
+
+/// Default port number used by Exonum for communication between nodes.
+pub const DEFAULT_EXONUM_LISTEN_PORT: u16 = 6333;
+
+/// Generate public and private configs of the node.
+#[derive(StructOpt, Debug, Serialize, Deserialize)]
+#[structopt(rename_all = "kebab-case")]
```

Comment:
New structopt version uses this option by default.
https://github.com/TeXitoi/structopt/blob/master/CHANGELOG.md#v030-2019-08-30